### DISCUSSION TITLE
Heroku Support

### DISCUSSION THREAD

**Discussion by amjiddader** [bluenviron/mediamtx#674] [Questions]
## Deploy on heroku

Is it posible to deploy on heroku? 
I understad heroku doesnot allow much open ports but we can!
Use PORT 80 for hls and PORT 443 for RTMP - or vice-virsa

We will deply this app on heroku either with docker with above 2 ports or we can deploy anything i.e python build pack just to init heroku container. 

-- PS: it will not allow input internal input streams. But to just relay streams. 
Also we can use variable to decide which port to use for which serivce and only allow 2 services./
Same way we can add relay stream from ENV i.e STREAM_URL : PATH

- Why am i asking -
- I have just 3 streams i want to relay ( proxy them and they need a US server )

Regards

**Comment by aler9** [comment]
Hello, just change the ports you need in the configuration file and you're good to go.

**Comment by amjiddader** [comment]
Can you create a repo that can be deployed to heroku .

I tried ( i never used heroku before)
Few days back i need local USA tv channels and channels are locked to US.
I read somewhere : Heroku server location is US

I just created a PHP repo and it was easy to use app.json heroku.yml for PHP app.


When i tried to deploy this i was not able to make this thing work on Heroku.
What i did : was not able to create valid app.json file ( i use different build packs and while deploy i got error
* CAN NOT DEPLOY BUILDPACK XXXX ON HEROKU-20) 

 -I changed ports to 80 for hls and 443 for rtmp 
- created a .sh file that would set permission and start ....
- But i never reach to that bash .sh file.

Can you try to deploy one on port 80 & 443

[a comment by an automated account is not shown]
